### ISSUE TITLE
Codegeneration fails with case sensitive data types in postgres 9.0 or less

### ISSUE THREAD

**Issue by jklingsporn** [jOOQ/jOOQ#9917]
### Expected behavior and actual behavior:
Prior to upgrading to jOOQ 3.13.1 (coming from 3.12.4) it was totally fine to have case sensitive data types (enum). After upgrading, the code generation fails saying the datatype was unkown (sorry for not having exact stack trace, but I fixed the issue before filing this issue by renaming the datatypes into snake case). I still know the error origin, though: `PostgresDatabase line 1025`. I suspect the error might be in the beginning of the method
```
private List<String> enumLabels(String nspname, String typname) {
        Field<Object> cast = field("{0}::{1}", PG_ENUM.ENUMLABEL, **name(nspname, typname)**);

        if (canCastToEnumType == null) {

            // [#2917] Older versions of PostgreSQL don't support the above cast
            try {
                canCastToEnumType = true;
                return enumLabels(nspname, typname, cast);
            }
            catch (DataAccessException e) {
                canCastToEnumType = false;
            }
        }

        return canCastToEnumType ? enumLabels(nspname, typname, cast) : enumLabels(nspname, typname, PG_ENUM.ENUMLABEL);
    }
```
which probably should be `quotedName(nspname, typname)` instead?

### Steps to reproduce the problem (if possible, create an MCVE: https://github.com/jOOQ/jOOQ-mcve):
Create a camel case data type and let the code generator do its work.

### Versions:

- jOOQ: 3.13.1
- Java: 1.8.131
- Database (include vendor): Postgres 10.5
- OS: Mac OSX
- JDBC Driver (include name if inofficial driver): postgresql::42.2.2

**Comment by lukaseder** [comment]
It seems this got fixed in the meantime for PostgreSQL 9.1+ via https://github.com/jOOQ/jOOQ/issues/10821, can you confirm?

**Comment by lukaseder** [comment]
Yeah, reverting the #10821 fix, I get that exception that you've mentioned:

> 14:07:16,083 [WARNING] SQL exception            : Exception while executing meta query: ERROR: type "public.u_9917_case_sensitive" does not exist Position: 370
> 
> If you think this is a bug in jOOQ, please report it here: https://github.com/jOOQ/jOOQ/issues/new
> 
> Note you can mute some exceptions using the configuration/onError flag
> 
> ```sql
> select pg_catalog.pg_enum.enumlabel
> from (
>   pg_catalog.pg_enum
>     join (
>       pg_catalog.pg_type as alias_108128242
>         join pg_catalog.pg_namespace as alias_61162684
>           on alias_108128242.typnamespace = alias_61162684.oid
>     )
>       on pg_catalog.pg_enum.enumtypid = alias_108128242.oid
>   )
> where (
>   alias_61162684.nspname = 'public'
>   and alias_108128242.typname = 'u_9917_Case_Sensitive'
> )
> order by pg_catalog.pg_enum.enumlabel::public.u_9917_Case_Sensitive```

**Comment by lukaseder** [comment]
The regression was probably introduced by #9511

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.16.0, 3.15.5 (https://github.com/jOOQ/jOOQ/issues/12617), 3.14.16 (https://github.com/jOOQ/jOOQ/issues/12618)
